src/main/java/org/openrewrite/java/testing/junit5/AddParameterizedTestAnnotation.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.junit5;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.tree.*;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AddParameterizedTestAnnotation extends Recipe {
+    private static final String PARAMS_PATH = "@org.junit.jupiter.params.provider.";
+    private static final AnnotationMatcher TEST_ANNOTATION_MATCHER = new AnnotationMatcher("@org.junit.jupiter.api.Test");
+    private static final AnnotationMatcher VALUE_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"ValueSource");
+    private static final AnnotationMatcher CSV_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"CsvSource");
+    private static final AnnotationMatcher METHOD_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"MethodSource");
+    private static final AnnotationMatcher NULL_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"NullSource");
+    private static final AnnotationMatcher EMPTY_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"EmptySource");
+    private static final AnnotationMatcher NULL_EMPTY_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"NullAndEmptySource");
+    private static final AnnotationMatcher ENUM_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"EnumSource");
+    private static final AnnotationMatcher CSV_FILE_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"CsvFileSource");
+    private static final AnnotationMatcher ARG_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"ArgumentsSource");
```

Comment:
Had you also considered something like this?
```suggestion
    private static final List<AnnotationMatcher> SOURCE_ANNOTATIONS = Stream.of(
                    "ArgumentsSource",
                    "CsvFileSource",
                    "CsvSource",
                    "EmptySource",
                    "EnumSource",
                    "MethodSource",
                    "NullAndEmptySource",
                    "NullSource",
                    "ValueSource")
            .map(annotation -> new AnnotationMatcher("@org.junit.jupiter.params.provider." + annotation))
            .collect(Collectors.toList());
```